### PR TITLE
fixing nimble

### LINKED ISSUE #8096: [Bug]: nimble fails to work after update
### Problem description

using nimble to compile a project results in the following error: (tested 2 times)

```
oserr.nim             raiseOSError
Error: unhandled exception: Permission Denied
/data/local/tmp/nimblecache-1604532304
```
(why is it accessing /data though. seems it has some configuration files that it is trying to access which was not
how it was.)

nimble v0.13.1 compiled at 2021-11-28
couldnt determine git hash

### What steps will reproduce the bug?

use nimble to compile a project, will return 1 with above output.

cd project/
nimble build

### What is the expected behavior?

it should compile correctly.

Verifying dependencies for project@VERSION
Building using c backend

### System information

termux-info:

```
Application version:
0.117
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://packages-cf.termux.org/apt/termux-main/ stable main
# game-repo (sources.list.d/game.list)
deb https://packages.termux.org/apt/termux-games games stable
# science-repo (sources.list.d/science.list)
deb https://packages.termux.org/apt/termux-science science stable
Updatable packages:
apt/stable 2.3.13-1 aarch64 [upgradable from: 2.3.13]
bash/stable 5.1.12-2 aarch64 [upgradable from: 5.1.12]
dialog/stable 1.3-20210621-1 aarch64 [upgradable from: 1.3-20210621-0]
git/stable 2.34.1 aarch64 [upgradable from: 2.34.0]
grep/stable 3.7-2 aarch64 [upgradable from: 3.7-1]
ndk-sysroot/stable 23b-2 aarch64 [upgradable from: 23b-1]
sed/stable 4.8-2 aarch64 [upgradable from: 4.8-1]
util-linux/stable 2.37.2 aarch64 [upgradable from: 2.37]
xxhash/stable 0.8.1 aarch64 [upgradable from: 0.8.0-1]
Android version:
8.1.0
Kernel build information:
Linux localhost 3.18.71-perf-g78568c2-dirty #1 SMP PREEMPT Fri Sep 6 14:44:47 CST 2019 aarch64 Android
Device manufacturer:
LENOVO
Device model:
Lenovo TB-8504X
```

### PR DESCRIPTION
**Pull request by sk0kanik**
added -d:tempDir: when compiling nimble and koch because host nim compiler is using /data/local/tmp as temporary directory. 
fixes #8096

### CODE REVIEW

File: packages/nim/build.sh [hunk 1]
```diff
@@ -42,9 +42,9 @@ termux_step_make() {
 	make LD=$CC uos=linux mycpu=$NIM_ARCH myos=android  -j $TERMUX_MAKE_PROCESSES useShPath=$TERMUX_PREFIX/bin/sh
 	cp config/nim.cfg ../host-build/config
 
-	nim --cc:clang --clang.exe=$CC --clang.linkerexe=$CC --opt:size --define:termux -d:release --os:android --cpu:$NIM_ARCH  -t:"$CPPFLAGS $CFLAGS" -l:"$LDFLAGS -landroid-glob" c koch.nim
+	nim --cc:clang --clang.exe=$CC --clang.linkerexe=$CC --opt:size --define:termux -d:release --os:android --cpu:$NIM_ARCH  -t:"$CPPFLAGS $CFLAGS" -l:"$LDFLAGS -landroid-glob" -d:tempDir:/data/data/com.termux/files/usr/tmp c koch.nim
```
Comment (Grimler91) [inline packages/nim/build.sh:45]: Please use $TERMUX_PREFIX instead of hardcoding the path

**Review by Grimler91 — CHANGES_REQUESTED** [review]

File: packages/nim/build.sh (hunk 1 above)
Comment (sk0kanik) [reply packages/nim/build.sh]: Done. Thanks.

**Comment by buttaface** [comment]
This wasn't fixed by #8097?

**Comment by sk0kanik** [comment]
> This wasn't fixed by #8097?

#8097 fixed getTempDir() on nim library. I was confident, it will also fix nimble. But later when I tested, I found that nimble and koch is compiled saparately by host nim compiler. Changes on #8097 only affect cross-compiled nim compiler and nimble still uses /data/local/tmp.

**Pull request merged**